### PR TITLE
Posts have user info

### PR DESCRIPTION
**Pull request by Aruzal**
Questions, Reviews, Comments now all have a user field that contains the publicProfile data of the user who made the post.

### CODE REVIEW

**Comment by NunoDasNeves** [comment]
All cards now receive and properly use user object, with the exception of the picture, which should replace the letter-in-circle default if present

**Comment by NunoDasNeves** [comment]
plus a bunch of random bug fixes and things...

**Comment by NunoDasNeves** [comment]
Also I prettified the user/comment/review/question data a bunch, building off Trav's work

File: backend/src/controllers/question.js [hunk 1]
```diff
@@ -10,8 +11,14 @@ exports.getQuestion = function ({ params }, res) {
     const getQuestion = Promise.all([
         questionModel.getQuestion(params.id),
         likesModel.getLikes({ type: 'question', id: params.id })
-    ])
-        .then(([question, likes]) => { return { ...question, ...likes } })
+    ]).then(([question, likes]) => {
+        return Promise.all([
```
[the remark this answers is not in the document]
Comment (Aruzal) [reply backend/src/controllers/question.js]: Just makes it easier to add stuff later on if required.

File: backend/src/controllers/course.js [hunk 2]
```diff
@@ -28,23 +29,25 @@ exports.getCourseQuestions = function ({ params, query }, res) {
...
+            Promise.all(questions.map(question => userModel.getPublicProfile(question.userID)))
+        ]).then(([likes, users]) => {
+            for (var i = 0; i < questions.length; i++) {
+                delete questions[i].userID
+                questions[i].likes = likes[i].likes
+                questions[i].user = users[i]
+            }
+            const lastPage = Math.trunc((questionCount[0]['COUNT()'] + pageSize) / pageSize)
+            return {
+                'meta': {
```
[the remark this answers is not in the document]
Comment (Aruzal) [reply backend/src/controllers/course.js]: Is this an issue that needs to be changed?

File: backend/src/controllers/course.js [hunk 3]
```diff
@@ -28,23 +29,25 @@ exports.getCourseQuestions = function ({ params, query }, res) {
...
-                return {
-                    'meta': {
-                        'curr': pageNumber,
-                        'last': lastPage,
-                        'pageSize': pageSize
-                    },
-                    'data': questions
-                }
-            })
+        return Promise.all([
```
[the remark this answers is not in the document]
Comment (Aruzal) [reply backend/src/controllers/course.js]: I was going to do this thought having a user key was nicer in terms of the json response. Same functionality though.

**Pull request merged**